app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -56,5 +56,25 @@ def new_token
         )
       end
     end
+
+    class Authenticator
+      # This delegates to all the work to the call method created automatically
+      # by CommandClass
+      #
+      # This is needed because we need `valid?` to exist on the Authenticator
+      # class, but that class contains only a metaprogramming generated
+      # `call(authenticator_input:)` method.  The methods we define in the
+      # block passed to `CommandClass` exist only on the private internal
+      # `Call` objects created each time `call` is run.
+      def valid?(input)
```

Comment:
I think we need a design meeting to understand the purpose of the function, i do not like it

Reply:
currently it is needed in order to be present in the `/authenticators` and `/info` endpoints. it is also needed to be present to authenticate in other authenticators but it may not be needed by the JWT authn.

i agree that we should think of a proper way to achieve the above for all the authenticators as it's a workaround that we are doing since OIDC already, from when @jonahx introduced the command class for authentication

Reply:
I opened a new pr (https://github.com/cyberark/conjur/pull/2195) that adds a new API endpoint for jwt authenticator, please review it and add your comment if its still relevant